django_project/frontend/src/pages/Dashboard/MapLibre/LayerType/RasterCog.jsx

```diff
@@ -56,26 +66,55 @@ export default function rasterCogLayer(map, id, data, setData, contextLayerData,
       if (!colors.length) {
         return
       }
+
+            const requestBody = {
+          url: data.url,
+          class_type: dynamic_classification,
+          class_num: dynamic_class_num,
+          colors: colors,
+      }
+
+      let classifications = [];
+
+      const cacheKey = await generateCacheKey(data.url, requestBody);
+      const cachedData = sessionStorage.getItem(cacheKey);
```

Comment:
You can put it on django backend cache instead in frontend (so other user can access the cache)